packages/smart-contracts/src/contracts/ERC20SwapToPay.sol

```diff
@@ -118,24 +150,51 @@ contract ERC20SwapToPay is Ownable {
     );
 
     // Give the change back to the payer, in both currencies (only spent token should remain)
-
     if (spentToken.balanceOf(address(this)) > 0) {
       spentToken.safeTransfer(msg.sender, spentToken.balanceOf(address(this)));
     }
+
     if (requestedToken.balanceOf(address(this)) > 0) {
       requestedToken.safeTransfer(msg.sender, requestedToken.balanceOf(address(this)));
     }
   }
 
+  /**
+   * Internal function to calculate the amount of request fee to pay with a transaction.
+   * @param _amount The Wei amount to calculate the fee from.
+   * @dev The _amount has to be nominated in Wei.
+   * @dev returns the _feeAmount in Wei. 
+   */
+  function _calculateSwapFee(uint256 _amount) internal view returns (uint256 _swapFeeAmount) {
+    _swapFeeAmount = (_amount.mul(currentSwapFee)).div(1000);
+  }
+
   /*
-  * Admin functions to edit the admin, router address or proxy address
+  * Admin functions to edit the router address or proxy address.
   */
-
   function setPaymentProxy(address _paymentProxyAddress) public onlyOwner {
     paymentProxy = IERC20FeeProxy(_paymentProxyAddress);
   }
 
   function setRouter(address _newSwapRouterAddress) public onlyOwner {
     swapRouter = IUniswapV2Router02(_newSwapRouterAddress);
   }
+
+  /** 
+   * Admin function to edit the swap fee address.
+   * @param _newFeeAddress Address that receives the swap fees.
+   */
+  function setFeeAddress(address _newFeeAddress) public onlyWhitelisted {
```

Comment:
any reason why this is not using `onlyOwner`?

Reply:
`onlyWhitelisted` was a requirement in the spec I got. 
The benefit is that It allows multiple addresses to be whitelisted instead of `onlyOwner`.